inc/c_util/slist.h

```diff
@@ -9,18 +9,31 @@
 #include <stdbool.h>
 #endif
 
-typedef struct SLIST_ENTRY_TAG
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+typedef struct SINGLYLINKEDLIST_ENTRY_TAG
 {
-    struct SLIST_ENTRY_TAG *next;
-} SLIST_ENTRY, *PSLIST_ENTRY;
+    struct SINGLYLINKEDLIST_ENTRY_TAG* next;
+} SINGLYLINKEDLIST_ENTRY, * PSINGLYLINKEDLIST_ENTRY;
+
+#define SLIST_IS_EMPTY_RESULT_VALUES \
+    EMPTY,               \
+    NOT_EMPTY,      \
+    INVALID_ARGS
+MU_DEFINE_ENUM(SLIST_IS_EMPTY_RESULT, SLIST_IS_EMPTY_RESULT_VALUES);
```

Comment:
Nitpick: Let's have a newline between the last line of SLIST_IS_EMPTY_RESULT_VALUES and the MU_DEFINE_ENUM.
The reason is readability and also that sometimes we leave a trailing \ so that we make it easier to add new values :-)